pyteal/ast/router.py

```diff
@@ -427,15 +428,21 @@ def add_method_handler(
             raise TealInputError(
                 f"registered method {method_signature} is never executed"
             )
-        oc_create: list[EnumInt] = call_configs.oc_under_call_config(CallConfig.CREATE)
-        oc_call: list[EnumInt] = call_configs.oc_under_call_config(CallConfig.CALL)
         if method_signature in self.added_method_sig:
             raise TealInputError(f"re-registering method {method_signature} detected")
         self.added_method_sig.add(method_signature)
 
         wrapped = Router.wrap_handler(True, method_call)
 
-        if any(str(OnComplete.ClearState) == str(x) for x in oc_create):
+        def partition(cc: CallConfig) -> Tuple[bool, list[EnumInt]]:
+            (not_clear_states, clear_states) = more_itertools.partition(
+                lambda x: str(x) == str(OnComplete.ClearState),
+                call_configs._oc_under_call_config(cc),
+            )
+            return (len(list(clear_states)) > 0, list(not_clear_states))
```

Comment:
```suggestion
            return (bool(clear_states), not_clear_states)
```

I believe this is equivalent